app/src/main/java/org/kiwix/kiwixmobile/history/HistoryActivity.java

```diff
@@ -111,15 +116,34 @@ protected void onCreate(Bundle savedInstanceState) {
       actionBar.setTitle(R.string.history);
     }
 
-
     historyAdapter = new HistoryAdapter(historyList, deleteList, this);
     recyclerView.setAdapter(historyAdapter);
+    layoutManager = new LinearLayoutManager(this, RecyclerView.VERTICAL, false);
+    recyclerView.setLayoutManager(layoutManager);
   }
 
   @Override
   protected void onResume() {
     super.onResume();
     presenter.loadHistory(sharedPreferenceUtil.getShowHistoryCurrentBook());
+    if (listState != null) {
+      layoutManager.onRestoreInstanceState(listState);
+    }
+  }
+
+  protected void onSaveInstanceState(Bundle state) {
+    super.onSaveInstanceState(state);
+    // Save list state
```

Comment:
It's good to have a well commented and well-documented code but it is not desirable to have comments on everything we do. In this case, the functionality is obvious from the name of the method. You should probably remove these comments.

Reply:
@abdulwd  Comments removed.